homeassistant/components/cert_expiry/__init__.py

```diff
@@ -27,15 +27,20 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     port = entry.data[CONF_PORT]
 
     coordinator = CertExpiryDataUpdateCoordinator(hass, host, port)
-    await coordinator.async_config_entry_first_refresh()
 
     hass.data.setdefault(DOMAIN, {})
     hass.data[DOMAIN][entry.entry_id] = coordinator
 
     if entry.unique_id is None:
         hass.config_entries.async_update_entry(entry, unique_id=f"{host}:{port}")
 
-    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+    async def async_finish_startup(_):
+        await coordinator.async_refresh()
+        hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    entry.async_on_unload(
+        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STARTED, async_finish_startup)
```

Comment:
How will this work when setting up the integration after Home Assistant has started?

Reply:
Good catch. Opened #56298 to handle both cases.